babble-backend/controllers/user-controller.js

```diff
@@ -0,0 +1,67 @@
+const db = require('../database/db');
+
+module.exports = {
+    changePassword,
+    changeEmail,
+    deleteAccount,
+    getSavedPosts,
+    getMyPosts,
+    deletePost
+}
+
+/*************************************** */
+/* Get the userid with res.locals.userid */
+/*************************************** */
+
+/* change user password, req.body: email, oldPassword, newPassword */
+function changePassword (req, res) {
+    // TODO
+    res.send('change password');
+}
+
+/* change user email, req.body: oldEmail, newEmail, password */
+function changeEmail (req, res) {
+    // TODO
+    res.send('change email');
+}
+
+/* delete account, req.body: email, password */
+function deleteAccount (req, res) {
+    // TODO
+    res.send('delete account');
+}
+
+/* get saved posts */
+function getSavedPosts (req, res) {
+    // TODO
+    res.send('got saved posts');
+}
+
+/* get posts the user made */
+function getMyPosts (req, res) {
+    // TODO
+    res.send('got my posts');
+}
+
+/* delete a post made the user, query: post id (pid) */
+/* suggestion: if the post exists and has already been deleted, 
```

Comment:
I disagree because the feed returns only non-deleted posts, so the client should have no knowledge of these posts existing after they've been deleted (i.e. 404 if they try to delete a post that doesn't "exist").

Reply:
Ok yeah, what you said makes more sense.